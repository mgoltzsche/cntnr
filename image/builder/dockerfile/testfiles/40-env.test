FROM alpine:3.7

ENV VAR1 myval
ENV VAR1 entrypoint.sh
ENV VARX=myvalx
ENV VARA=80 VARB=443
ENV VARL="long value"


# env substitution:

ENV SVAR1=$VAR1
ENV SVAR2=${VAR1}
ENV SVAR3=\$VAR1
ENV SVAR4=\${VAR1}

ENV USR 0:0
ENV BASEIMAGE alpine:3.7
ENV WORK /etc

COPY $VAR1 /dest
COPY /entrypoint.sh /$VAR1
COPY --chown=$USR /entrypoint.sh /dest
COPY --from=$BASEIMAGE etc/hosts /dest

ADD $VAR1 /dest
ADD /entrypoint.sh /$VAR1
ADD --chown=$USR /entrypoint.sh /dest

EXPOSE $VARA $VARB

LABEL $VARX=$VARL

ENV VARSTOP SIGTERM
STOPSIGNAL $VARSTOP

USER $USR

VOLUME $VARX $VARL

WORKDIR $WORK
WORKDIR /

# require no substitution:
RUN [ "$VAR1" = entrypoint.sh ]
ENTRYPOINT echo $VAR1
ENTRYPOINT [ "/bin/echo", "$VAR1" ]
CMD echo $VAR1
CMD [ "/bin/echo", "$VAR1" ]

# imagebuidler: change bundle state after RUN initialized bundle already
ENV NEWVAR $VARX
WORKDIR $WORK

# ASSERT CFG /config/User=0:0
# ASSERT RUN [ "$NEWVAR" = myvalx ] || (echo env not reapplied to bundle, NEWVAR=$NEWVAR, VAR1=$VAR1; false)
# ASSERT RUN [ "$(pwd)" = /etc ] || (echo workdir not reapplied to bundle, pwd: "$(pwd)"; false)