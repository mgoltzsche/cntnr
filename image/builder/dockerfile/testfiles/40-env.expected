FROM alpine:3.7
NAME 0

ENV "VAR1"="myval"
ENV "VAR1"="entrypoint.sh"
ENV "VARX"="myvalx"
ENV "VARA"="80" "VARB"="443"
ENV "VARL"="long value"


ENV "SVAR1"="entrypoint.sh"
ENV "SVAR2"="entrypoint.sh"
ENV "SVAR3"="$VAR1"
ENV "SVAR4"="${VAR1}"

ENV "USR"="0:0"
ENV "BASEIMAGE"="alpine:3.7"
ENV "WORK"="/etc"

COPY dir="./ctx" "entrypoint.sh" "/dest" nil
COPY dir="./ctx" "/entrypoint.sh" "/entrypoint.sh" nil
COPY dir="./ctx" "/entrypoint.sh" "/dest" 0:0
COPY image="alpine:3.7" "etc/hosts" "/dest" nil

ADD dir="./ctx" "entrypoint.sh" "/dest" nil
ADD dir="./ctx" "/entrypoint.sh" "/entrypoint.sh" nil
ADD dir="./ctx" "/entrypoint.sh" "/dest" 0:0

EXPOSE 80 443

LABEL "myvalx"="long value"

ENV "VARSTOP"="SIGTERM"
STOPSIGNAL SIGTERM

USER 0:0

VOLUME "myvalx" "long value"

WORKDIR /etc
WORKDIR /


RUN "/bin/sh" "-c" "[ \"$VAR1\" = entrypoint.sh ]"
ENTRYPOINT "/bin/sh" "-c" "echo $VAR1"
ENTRYPOINT "/bin/echo" "$VAR1"
CMD "/bin/sh" "-c" "echo $VAR1"
CMD "/bin/echo" "$VAR1"

ENV "NEWVAR"="myvalx"
WORKDIR /etc